include/circt/Dialect/HW/HWAttributesNaming.td

```diff
@@ -16,16 +16,28 @@ def InnerRefAttr : AttrDef<HWDialect, "InnerRef"> {
     This works like a symbol reference, but to a name inside a module.
   }];
   let mnemonic = "innerNameRef";
-  let parameters = (ins "::mlir::StringAttr":$module, "::mlir::StringAttr":$name);
+  let parameters = (ins "::mlir::StringAttr":$module, "::mlir::StringAttr":$name,
+                        "unsigned":$fieldID);
```

Comment:
I wonder it is possible to make "fieldID" to optional parameter whose default value is zero.